ternimal.rs

```diff
@@ -251,6 +253,10 @@ fn main() {
         // Hide cursor while printing canvas to avoid flickering
         print!("\x1B[?25l{}\x1B[?25h\n", output);
 
+        if duration >= 0.0 && time > duration {
```

Comment:
Moved the check (and the cursor reset by necessity) to the beginning of the loop, only downside is this will cause a duration that's greater than 0 but very small to not draw any frames, which feels a bit wrong.

Reply:
I don't see an issue with this: If we have already spent more time than our budget, we need to exit. That part is exactly as it should be IMO.

What I don't like is that a duration of zero is accepted, which never makes sense. The default duration should be zero, meaning "run forever".